File: 09Scala/Ex1_6.scala
import scala.util.control._

object Ex1_6 {
  def main(args: Array[String]): Unit = {
    val initialDivisor = 5
    var totalIterations = 0
    for (i <- 1 to 5) {
      var dividend = i
      var divisor = initialDivisor
      var iterationsToSolveCount = 0

      val loop = new Breaks;
      loop.breakable {
        while (1 == 1) {
          iterationsToSolveCount += 1
          var remainder = dividend % divisor
          if (remainder == 0) {
            loop.break;
          }
          dividend = divisor
          divisor = remainder
        }
      }

      totalIterations += iterationsToSolveCount

    }

    val result = totalIterations.toFloat / initialDivisor

    println(s"$result")
  }
}

File: 08Ruby/ex1_6.rb
class App
  def main
    initialDivisor = 5
    totalIterations = 0
    for i in 1..5
      dividend = i
      divisor = initialDivisor
      iterationsToSolveCount = 0

      while 1 == 1 do
        iterationsToSolveCount += 1
        remainder = dividend % divisor
        if remainder == 0 then break end
        dividend = divisor
        divisor = remainder
      end

      totalIterations += iterationsToSolveCount
    end

    result = Float(totalIterations) / initialDivisor
    puts "#{result}"
    
  end
end

if __FILE__ == $0
  app = App.new
  app.main
end
